test/contracts/HCElection.aes

```diff
@@ -125,7 +145,20 @@ main contract HCElection =
     pick_validator(n, (_, stake) :: validators)            = pick_validator(n - stake, validators)
 
   function assert_protocol_call() =
-      require(Call.caller == Contract.creator, "Must be called by the protocol")
+    require(Call.caller == Contract.creator, "Must be called by the protocol")
+
+  function calc_carry_over(carry_over_flag : bool) =
```

Comment:
You can match in the function header, something like this is possible:
```
function 
  calc_carry_over : bool => int
  calc_carry_over(false) = 0
  calc_carry_over(true) = state.pin_reward.base + state.pin_reward.carry_over
```

Reply:
Oh, that is nice. Did not realize that. Will change to that, mostly because it's cool. Don't _really_ know if it adds to the readability of the code (anywhere in any language really).